python/kfserving/kfserving/storage.py

```diff
@@ -78,6 +82,32 @@ def _download_gcs(uri, temp_dir: str):
                     os.makedirs(local_object_dir, exist_ok=True)
             blob.download_to_filename(os.path.join(temp_dir, subdir_object_key))
 
+    @staticmethod
+    def _download_blob(uri, temp_dir: str):
+        m = re.search(_BLOB_RE, uri)
+        account_name = m.group(1)
+        storage_url = m.group(2)
+        container_name, blob_url = storage_url.split("/", 1)
+
+        block_blob_service = BlockBlobService(account_name=account_name)
+
+        blobs = block_blob_service.list_blobs(container_name, prefix=blob_url)
+
+        for blob in blobs:
+            print(blob.name)
+            if "/" in "{}".format(blob.name):
+                head, tail = os.path.split("{}".format(blob.name))
+                file_path = temp_dir + "/" + head + "/" + tail
+                dir_path = temp_dir + "/" + head
+
+                if (os.path.isdir(dir_path)):
+                    block_blob_service.get_blob_to_path(container_name, blob.name, file_path)
+                else:
+                    os.makedirs(dir_path, exist_ok=True)
```

Comment:
If you have exist_ok=True, why do you need the if check? Cleaner to just do without?

Reply:
unfortunately we need to create directory.